### ISSUE TITLE
Telegraf wrapper for Nestjs

### ISSUE THREAD

**Issue by AldeonMoriak** [telegraf/telegraf#1094]
Hello, can you make a wrapper for Nestjs?

**Comment by bukhalo** [comment]
Hello, @AldeonMoriak you can try my module [NestJS Telegraf](https://github.com/bukhalo/nestjs-telegraf). At the moment it contains some bugs (mainly due to poor typing in Telegaf), but is used by some people in production. If you have any questions, please create a new issue.

**Comment by Siedlerchr** [comment]
I am using Telegaf with Nest.js out of the box:

I have craeted a simple service:
```
@Injectable()
export class TelegramService {

    TIMEOUT_MILLI_SECONDS: number;
    bot: Telegraf<MyContext>;
    constructor(configService: ConfigService, private httpService: HttpService)) {

        const token = configService.get<string>('TELEGRAM_BOT_TOKEN')
        const timeoutInSeconds = configService.get<number>("TIMEOUT_SECONDS") || 60

        this.TIMEOUT_MILLI_SECONDS = timeoutInSeconds * 1000;

        /** @type {typeof import('telegraf').Telegraf} */
        const Telegraf: typeof import('telegraf').Telegraf = require('telegraf')
        this.bot = new Telegraf(token);
        this.bot.use(localSession.middleware());
        console.log("ctor", this.bot);

        this.initBotHandling(); handle 
    }
```
Add that Service Class to the providers Array in the app.module
And in the main.ts init it:

```
async function bootstrap() {

  const app = await NestFactory.create(AppModule);

  const telegramService = app.get(TelegramService);
  const bot = telegramService.bot;
  const isDev = process.env.NODE_ENV === 'development'
  if (isDev) {
    bot.startPolling()
    bot.telegram.deleteWebhook();
  }
  else {
    app.use(bot.webhookCallback("/"))
    bot.telegram.setWebhook(process.env.DOMAIN) //your domain

  }
  const port = process.env.PORT || 3000;
  await app.listen(port)

}
bootstrap();

```

**Comment by KnorpelSenf** [comment]
We should maybe consider adding this to the docs after #1205. @wojpawlik?

**Comment by wojpawlik** [comment]
@bukhalo please link to https://nestjs-telegraf.vercel.app/docs/bot-injection from NestJS Telegraf's readme. Stages aren't the only reason to wish to use `Telegraf` directly. For example, someone might be unwilling to refactor their already working bot.

Then, while I'll be adding links to Telegraf 4.x compatible packages to readme, I'll link NestJS Telegraf as well.

**Comment by evilsprut** [comment]
@wojpawlik `nestjs-telegraf` v2.0.0 (based on `telegraf` v4) is still in beta. This is mainly because the documentation is not ready yet. As soon as I make the new documentation and publish the release version, I'll write here. Thx.

**Comment by evilsprut** [comment]
@wojpawlik I released version 2.0.0 and slightly updated the documentation. If you think something else needs to be added, please let me know.

**Comment by wojpawlik** [comment]
> @bukhalo please link to https://nestjs-telegraf.vercel.app/docs/extras/bot-injection from NestJS Telegraf's readme. Stages aren't the only reason to wish to use `Telegraf` directly. For example, someone might be unwilling to refactor their already working bot.

Also, the `"homepage"` in `package.json` should probably be changed to https://nestjs-telegraf.vercel.app/.

Open a PR adding `nestjs-telegraf` to https://github.com/telegraf/telegraf#webhooks list after #1368 is merged.
